pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -130,36 +301,60 @@ func (ssc *defaultStatefulSetControl) UpdateStatefulSet(set *apps.StatefulSet, p
 	for i := range replicas {
 		// delete and recreate failed pods
 		if isFailed(replicas[i]) {
-			glog.V(2).Infof("StatefulSet %s is recreating failed Pod %s", set.Name, replicas[i].Name)
+			glog.V(4).Infof("StatefulSet %s is recreating failed Pod %s", set.Name, replicas[i].Name)
 			if err := ssc.podControl.DeleteStatefulPod(set, replicas[i]); err != nil {
-				return err
+				return &status, err
```

Comment:
We need to delete any failed Pods. This doesn't count as a termination. You should look at the deletion handler for Pods in the api server to understand it fully. When you delete a failed Pod you will just remove it from storage. Its already terminated. All our other deletes will set the termination grace and we have to wait for the Pod to turn down.